flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -334,6 +335,14 @@ extends Actor with ActorLogMessages with ActorSynchronousLogging {
           }
         }
 
+      /**
+       * Pin a ResultPartition corresponding to an IntermediateResultPartition
+       */
+      case LockResultPartition(partitionID, numConsumers) =>
+        val partitionManager: ResultPartitionManager = this.network.getPartitionManager
```

Comment:
@StephanEwen after a disconnect from the JM, the TM will clear the network stack components and the partition manager might be null. Is it possible that a msg is received after the disconnect, e.g. do we have to check for null here?